addons/binding/org.openhab.binding.km200/src/main/java/org/openhab/binding/km200/handler/KM200ThingHandler.java

```diff
@@ -0,0 +1,451 @@
+        Class<?> commandType = command.getClass();
+
+        if (commandType.isAssignableFrom(DateTimeType.class) || commandType.isAssignableFrom(DecimalType.class)
+                || commandType.isAssignableFrom(StringType.class) || commandType.isAssignableFrom(OnOffType.class)) {
+            gateway.prepareMessage(this.getThing(), channel, command);
+        } else if (commandType.isAssignableFrom(RefreshType.class)) {
+            gateway.refreshChannel(channel);
+        } else {
+            logger.warn("Unsupported Command: {} Class: {}", command.toFullString(), command.getClass());
+        }
+    }
+
+    /**
+     * Choose a tag for a channel
+     *
+     */
+    Set<String> checkTag(String unitOfMeasure, Boolean readOnly) {
+        Set<String> tags = new HashSet<String>();
+        if (unitOfMeasure.indexOf("°C") == 0 || unitOfMeasure.indexOf("K") == 0) {
+            if (readOnly) {
+                tags.add("CurrentTemperature");
+            } else {
+                tags.add("TargetTemperature");
+            }
+        }
+        return tags;
+    }
+
+    /**
+     * Choose a category for a channel
+     *
+     */
+    String checkCategory(String unitOfMeasure, String topCategory, Boolean readOnly) {
+        String category = null;
+        if (unitOfMeasure.indexOf("°C") == 0 || unitOfMeasure.indexOf("K") == 0) {
+            if (unitOfMeasure.indexOf("°C") == 0 || unitOfMeasure.indexOf("K") == 0) {
+                if (readOnly) {
+                    category = "Temperature";
+                } else {
+                    category = "Heating";
+                }
+            }
+        } else if (unitOfMeasure.indexOf("kW") == 0 || unitOfMeasure.indexOf("kWh") == 0) {
+            category = "Energy";
+        } else if (unitOfMeasure.indexOf("l/min") == 0 || unitOfMeasure.indexOf("l/h") == 0) {
+            category = "Flow";
+        } else if (unitOfMeasure.indexOf("Pascal") == 0 || unitOfMeasure.indexOf("bar") == 0) {
+            category = "Pressure";
+        } else if (unitOfMeasure.indexOf("rpm") == 0) {
+            category = "Flow";
+        } else if (unitOfMeasure.indexOf("mins") == 0 || unitOfMeasure.indexOf("minutes") == 0) {
+            category = "Time";
+        } else if (unitOfMeasure.indexOf("kg/l") == 0) {
+            category = "Oil";
+        } else if (unitOfMeasure.indexOf("%%") == 0) {
+            category = "Number";
+        } else {
+            category = topCategory;
+        }
+        return category;
+    }
+
+    /**
+     * Creates a new channel
+     *
+     */
+    Channel createChannel(ChannelTypeUID channelTypeUID, ChannelUID channelUID, String root, String type,
+            String currentPathName, String description, String label, Boolean addProperties, Boolean switchProgram,
+            StateDescription state, String unitOfMeasure) {
+        Channel newChannel = null;
+        Map<String, String> chProperties = new HashMap<>(1);
```

Comment:
Better not set a size or the correct size, now it will most likely need to resize on line 184

Reply:
OK, done.